cmd/qat_plugin/qat_plugin_test.go

```diff
@@ -362,3 +363,56 @@ func TestScanPrivate(t *testing.T) {
 		}
 	}
 }
+func eleInSlice(a string, list []string) bool {
+	for _, b := range list {
+		if b == a {
+			return true
+		}
+	}
+	return false
+}
+func TestPostAllocate(t *testing.T) {
+	response := new(pluginapi.AllocateResponse)
+	cresp := new(pluginapi.ContainerAllocateResponse)
+	response.ContainerResponses = append(response.ContainerResponses, cresp)
+	testMap := map[string]string{
+		"QAT29": "03:04.1",
+		"QAT13": "03:04.2",
+		"QAT6":  "03:04.3",
+		"QAT21": "03:04.4",
+	}
+	response.ContainerResponses[0].Envs = testMap
+	resultKey := []string{
+		"QAT0",
+		"QAT1",
+		"QAT2",
+		"QAT3",
+	}
+	expectedValues := map[string]struct{}{
+		"03:04.1": {},
+		"03:04.2": {},
+		"03:04.3": {},
+		"03:04.4": {},
+	}
+	dp := &devicePlugin{}
+	dp.PostAllocate(response)
+	if len(response.ContainerResponses[0].Envs) != 4 {
+		t.Fatal("Set wrong number of Environment Variables")
+	}
+
+	for k := range response.ContainerResponses[0].Envs {
+		if !eleInSlice(k, resultKey) {
+			t.Fatalf("Set wrong key: %s. The key should be in the range %v", k, resultKey)
+		}
+	}
+
+	for _, key := range resultKey {
+		if value, ok := response.ContainerResponses[0].Envs[key]; ok {
+			if _, ok := expectedValues[value]; ok {
+				delete(expectedValues, value)
+			} else {
+				t.Errorf("Unexpected value %s", value)
+			}
+		}
```

Comment:
It wouldn't harm to add one more `else` clause here reporting about not found expected `key` from the `resultKey` slice. But let's have it added in a follow-up PR.